tools/java/org/unicode/cldr/util/XPathParts.java

```diff
@@ -527,7 +562,14 @@ public XPathParts removeAttributes(String elementName, Collection<String> attrib
     }
 
     public XPathParts removeAttributes(int elementIndex, Collection<String> attributeNames) {
-        elements.get(elementIndex >= 0 ? elementIndex : elementIndex + size()).removeAttributes(attributeNames);
+        elementIndex = elementIndex >= 0 ? elementIndex : elementIndex + size();
+        if (elements.get(elementIndex).attributes == null || attributeNames == null) {
```

Comment:
change to:

Map<...> ea = elements.get(elementIndex).attributes;

if (temp == null || attributeNames == null || attributeNames.isEmpty() || Collections.disjoint(attributeNames, ea.keySet())) {

That verifies that we have to make the element mutable before going further.